pipelines/main/launch_upload_jobs.yml

```
@@ -23,6 +23,10 @@ steps:
           export BUILDKITE_PLUGIN_CRYPTIC_BASE64_SIGNED_JOB_ID_SECRET
           export ALLOW_FAIL="false"
 
+          # allow test-failures for non-release/tags so that nightly & PR builds
+          # can be uploaded to s3 even if tests are not passing
+          # (allowing for usage with juliaup)
+          export ALLOW_TEST_FAILURES=$(([[ "${BUILDKITE_BRANCH?}" == release-* ]] || [[ "${BUILDKITE_TAG:-}" == v* ]]) && echo "false" || echo "true")
```

Also, can we rename this something else? In Buildkite, "allow fail" (or "soft fail") means something different - it means failures show up as a red warning sign (not a red X), and failures don't block `Test` from being green.

Here, we have a different meaning, so can we make the variable name more unambiguous?

```suggestion
          export UPLOAD_EVEN_IF_TEST_FAILURES=$(([[ "${BUILDKITE_BRANCH?}" == release-* ]] || [[ "${BUILDKITE_TAG:-}" == v* ]]) && echo "false" || echo "true")
```